src/TimeSteppers/quasi_adams_bashforth_2.jl

```diff
@@ -52,9 +53,17 @@ function time_step!(model::AbstractModel{<:QuasiAdamsBashforth2TimeStepper}, Δt
 
     # Shenanigans for properly starting the AB2 loop with an Euler step
     euler = euler || (Δt != model.timestepper.previous_Δt)
+    
     χ = ifelse(euler, convert(eltype(model.grid), -0.5), model.timestepper.χ)
 
-    euler && @debug "Taking a forward Euler step."
+    if euler
+        @debug "Taking a forward Euler step."
+        # Ensure zeroing out all previous tendency fields to avoid errors in
+        # case G⁻ includes NaNs. See https://github.com/CliMA/Oceananigans.jl/issues/2259
+        for field in model.timestepper.G⁻
+            fill!(parent(field), 0)
```

Comment:
```suggestion
            field .= 0
```

I guess the code I suggested doesn't work on the cubed sphere. We can try the above! We might need to fix broadcasting but I can do that.